lib/membrane/timestamp_queue.ex

```diff
@@ -275,112 +290,104 @@ defmodule Membrane.TimestampQueue do
   """
   @spec pop_batch(t()) :: {[Action.resume_auto_demand()], [popped_value() | :none], t()}
   def pop_batch(%__MODULE__{} = timestamp_queue) do
-    {batch, timestamp_queue} = do_pop_batch(timestamp_queue)
+    do_pop_batch(timestamp_queue)
+  end
 
-    {actions, timestamp_queue} =
-      batch
-      |> Enum.reduce(MapSet.new(), fn
-        {pad_ref, {:buffer, _buffer}}, map_set -> MapSet.put(map_set, pad_ref)
-        _other, map_set -> map_set
-      end)
-      |> Enum.reduce({[], timestamp_queue}, fn pad_ref, {actions_acc, timestamp_queue} ->
-        {actions, timestamp_queue} = actions_after_popping_buffer(timestamp_queue, pad_ref)
-        {actions ++ actions_acc, timestamp_queue}
-      end)
+  defp do_pop_batch(%__MODULE__{} = timestamp_queue, actions_acc \\ [], items_acc \\ []) do
```

Comment:
I'd avoid default values in recursive functions, as it's easy to reset an accumulator by accident